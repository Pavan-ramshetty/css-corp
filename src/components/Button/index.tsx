import React, { ComponentProps, memo } from 'react';
import cn from 'classnames';

type ButtonProps = {
  children: string;
  icon?: any;
};

export type BtnProps = ButtonProps &
  Omit<ComponentProps<'button'>, keyof ButtonProps>;

const Button = ({ children, icon: Icon, ...props }: BtnProps) => {
  return (
    <button
      {...props}
      className="group relative w-full flex justify-center py-2 px-4 border border-transparent text-sm font-medium rounded-md text-white bg-indigo-600  focus:outline-none focus:ring-2 focus:ring-offset-2 hover:bg-indigo-700 focus:ring-indigo-500 disabled:bg-gray-500"
    >
      {Icon && (
        <span
          data-testid="icon-container"
          className="absolute left-0 inset-y-0 flex items-center pl-3"
        >
          <Icon fill="rgb(99 102 241 / 1)" />
        </span>
      )}
      {children}
    </button>
  );
};

export default memo(Button);
