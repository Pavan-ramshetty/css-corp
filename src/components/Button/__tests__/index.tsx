import * as React from 'react';
import { render, screen, fireEvent } from '@testing-library/react';
import LockIcon from 'assets/icons/lock.svg';
import Button from '../index';

const mockFn = jest.fn();

// beforeAll(() => {
//   // start server
// });

// beforeEach(() => {
//   console.log('before each');
//   // create mock methods
// });

// afterEach(() => {
//   // clear mock methods
//   jest.clearAllMocks();
// });

// afterAll(() => {
//   // stop server
// });

describe('"positive scenarios', () => {
  // define my postitive scenario data here
  beforeEach(() => {
    console.log('Positive scenario before all');
  });

  it('button snapshot', () => {
    const { container } = render(<Button>Sign In</Button>);
    expect(container.firstChild).toMatchSnapshot();
  });

  test('should first', () => {
    render(<Button>Sign In</Button>);
    const btn = screen.queryByRole('button');
    expect(btn).not.toBeNull();
    if (btn) {
      expect(btn.innerHTML).toMatch(/Sign In/i);
    }
  });

  test('should disable button', () => {
    render(<Button disabled>Sign In</Button>);
    const btn = screen.queryByRole('button');
    if (btn) {
      expect(btn).toBeDisabled();
    }
  });

  test('should icon visible', () => {
    render(<Button icon={LockIcon}>Sign In</Button>);
    const iconContainer = screen.queryByTestId('icon-container');
    expect(iconContainer).not.toBeNull();
  });

  test('should click button', () => {
    render(<Button onClick={mockFn}>Sign In</Button>);
    const btn = screen.queryByRole('button');
    if (btn) {
      fireEvent.click(btn);
      expect(mockFn).toBeCalledTimes(1);
    }
  });

  test('should click button', () => {
    render(<Button onClick={mockFn}>Sign In</Button>);
    const btn = screen.queryByRole('button');
    if (btn) {
      fireEvent.click(btn);
      expect(mockFn).toBeCalledTimes(1);
    }
  });
});

describe('negative scenarios', () => {
  afterAll(() => {
    console.log('negative scenarios after all');
  });
  test('should sum 2 + 2 = 4', () => {
    expect(2 + 2).toBe(4);
  });
});
