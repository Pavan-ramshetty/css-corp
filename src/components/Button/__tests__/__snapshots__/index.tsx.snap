// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`"positive scenarios button snapshot 1`] = `
<button
  class="group relative w-full flex justify-center py-2 px-4 border border-transparent text-sm font-medium rounded-md text-white bg-indigo-600  focus:outline-none focus:ring-2 focus:ring-offset-2 hover:bg-indigo-700 focus:ring-indigo-500 disabled:bg-gray-500"
>
  Sign In
</button>
`;
